src/main/kotlin/com/mdrobnak/lalrpop/inspections/NamedSymbolsInspection.kt

```diff
@@ -0,0 +1,55 @@
+package com.mdrobnak.lalrpop.inspections
+
+import com.intellij.codeInspection.*
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.util.PsiTreeUtil
+import com.mdrobnak.lalrpop.psi.*
+import com.mdrobnak.lalrpop.psi.ext.isExplicitlySelected
+import com.mdrobnak.lalrpop.psi.ext.isNamed
+import com.mdrobnak.lalrpop.psi.ext.removeName
+
+/**
+ * Inspection related to where named symbols can appear.
+ * Suggests removing the name if there are issues.
+ */
+object NamedSymbolsInspection : LocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : LpVisitor() {
+            override fun visitSymbol(symbol: LpSymbol) {
+                if (symbol.isNamed) {
```

Comment:
nit: To reduce nesting, try inverting this if statement and returning early if the symbol is not named.